maven-plugin-tools-annotations/src/main/java/org/apache/maven/tools/plugin/extractor/annotations/scanner/DefaultMojoAnnotationsScanner.java

```diff
@@ -85,12 +87,18 @@ public Map<String, MojoAnnotatedClass> scan( MojoAnnotationsScannerRequest reque
             for ( Artifact dependency : request.getDependencies() )
             {
                 scan( mojoAnnotatedClasses, dependency.getFile(), request.getIncludePatterns(), dependency, true );
+                if ( request.getMavenApiVersion() == null
+                     && dependency.getFile().getName().contains( "maven-plugin-api" ) )
+                {
+                    request.setMavenApiVersion( getSpecificationVersionOfJar( dependency.getFile() ) );
+                }
```

Comment:
Why not detect on Mojo level from dependencies?

Reply:
I don’t understand. This value is actually derived from the maven-plugin-api dependency. Per plugin there is only one classpath and dependency so this cannot differ per mojo…

Reply:
ok, other question

why not use 

```
                dependency.getArtifactId();
                dependency.getVersion();
```

instead of checking artifact file name and next read content from artifact manifest.

Reply:
Yes, a lot simpler, done in https://github.com/apache/maven-plugin-tools/pull/171/commits/4c8e2499919b75c2e1cd5a12e474dc65070f7d4f.